var/spack/repos/builtin/packages/groff/package.py

```diff
@@ -16,41 +16,62 @@ class Groff(AutotoolsPackage, GNUMirrorPackage):
+    version('1.22.3', sha256='3a48a9d6c97750bfbd535feeb5be0111db6406ddb7bb79fc680809cda6d828a5')
+
     # TODO: add html variant, spack doesn't have netpbm and its too
     # complicated for me to find out at this point in time.
     # See brew scripts for groff for guidance:
     # https://github.com/Homebrew/homebrew-core/blob/master/Formula/groff.rb
     # Seems troublesome...netpbm requires groff?
     variant('pdf', default=True, description='Build the `gropdf` executable.')
+    variant('x', default=False, description='Enable set of graphical options')
+    variant('uchardet', default=True,
+            description='Builds preconv with uchardet library for '
+                        'automatic file encoding detection')
 
     depends_on('gawk',  type='build')
     depends_on('gmake', type='build')
     depends_on('sed',   type='build')
     depends_on('ghostscript', when='+pdf')
+    # iconv is being asked whatever the release
+    depends_on('iconv', type=('build', 'link'))
+    # makeinfo is being searched for
+    depends_on('texinfo', type='build', when='@1.22.4:')
+    # configure complains when there is no uchardet that enhances preconv
+    depends_on('uchardet', type=('build', 'link'), when='@1.22.4:')
 
-    version('1.22.3', sha256='3a48a9d6c97750bfbd535feeb5be0111db6406ddb7bb79fc680809cda6d828a5')
-
-    # https://savannah.gnu.org/bugs/index.php?43581
-    # TODO: figure out why this patch does not actually work for parallel
-    # builds reliably.
-    # patch('gropdf.patch')
-    parallel = False
+    depends_on('libice', when='+x')
+    depends_on('libxaw', when='+x')
+    depends_on('libxmu', when='+x')
+    depends_on('libx11', when='+x')
 
     # The perl interpreter line in scripts might be too long as it has
     # not been transformed yet. Call scripts with spack perl explicitly.
-    patch('BuildFoundries.patch')
-    patch('pdfmom.patch')
+    patch('BuildFoundries.patch', when='@1.22.3')
+    patch('pdfmom.patch', when='@1.22.3')
 
     executables = ['^groff$']
 
+    @property
+    def parallel(self):
+        return self.spec.satisfies('@1.22.4')
+
     @classmethod
     def determine_version(cls, exe):
         output = Executable(exe)('--version', output=str, error=str)
         match = re.search(r'GNU groff version\s+(\S+)', output)
         return match.group(1) if match else None
 
     def configure_args(self):
-        args = [
-            "--without-x"
-        ]
+        args = ['--disable-silent-rules']
+        args.extend(self.with_or_without('x'))
+        if '@1.22.4:' in self.spec:
+            args.extend(self.with_or_without('uchardet'))
+        args.append('--with-libiconv-prefix={0}'.format(self.spec['iconv'].prefix))
         return args
+
+    @when('+x')
+    def setup_run_environment(self, env):
+        dir = join_path(self.prefix.lib, 'X11', 'app-defaults')
+        env.set_path('XFILESEARCHPATH', dir)
```

Comment:
If wwe don't use the `when` decorator we'll be more flexible in the future in case more unconditional logic needs to be added:
```suggestion
    def setup_run_environment(self, env):
        if self.spec.satisfies('+x'):
            dir = join_path(self.prefix.lib, 'X11', 'app-defaults')
            env.set_path('XFILESEARCHPATH', dir)
```